shared/src/main/scala/mlscript/TyperDatatypes.scala

```diff
@@ -31,6 +31,36 @@ abstract class TyperDatatypes extends TyperHelpers { self: Typer =>
     def rigidify(implicit lvl: Int): SimpleType = freshenAbove(level, body, rigidify = true)
   }
   
+  class MethodType(level: Int, body: SimpleType, val prts: List[TypeName], val single: Bool)
```

Comment:
I'm also uneasy about this being a subclass of `PolymorphicType`. Typing environments never map to this, right? Maybe a different base trait/class can be used for the purposes of method type handling.

Reply:
Addressed in d7e4fecb249181bed9b09864e3ca35b645bbd7d8